src/app/browser/qgsinbuiltdataitemproviders.cpp

```diff
@@ -1730,12 +1730,21 @@ void QgsDatabaseItemGuiProvider::populateContextMenu( QgsDataItem *item, QMenu *
           if ( conn2->capabilities().testFlag( QgsAbstractDatabaseProviderConnection::Capability::Schemas ) )
           {
             // Ok, this is gross: we lack a connection API for quoting properly...
-            sql = QStringLiteral( "SELECT * FROM %1.%2 LIMIT 10" ).arg( QgsSqliteUtils::quotedIdentifier( item->parent()->name() ), QgsSqliteUtils::quotedIdentifier( tableName ) );
+            sql = QStringLiteral( "FROM %1.%2" ).arg( QgsSqliteUtils::quotedIdentifier( item->parent()->name() ), QgsSqliteUtils::quotedIdentifier( tableName ) );
           }
           else
           {
             // Ok, this is gross: we lack a connection API for quoting properly...
-            sql = QStringLiteral( "SELECT * FROM %1 LIMIT 10" ).arg( QgsSqliteUtils::quotedIdentifier( tableName ) );
+            sql = QStringLiteral( "FROM %1" ).arg( QgsSqliteUtils::quotedIdentifier( tableName ) );
+          }
+
+          if ( conn2->providerKey() == QStringLiteral( "mssql" ) )
```

Comment:
Do you think it's worth moving this logic to the QgsAbstractDatabaseProviderConnection class itself? E.g. something like:

     /**
       * Returns a result size limited SQL query string generated for the given \a schema and table \a name, retrieving all columns for the first \a rows.
       *
       * The base class method returns the SQL query "SELECT * FROM table LIMIT 10". Subclasses may return database specific equivalents to this query.
      */
     virtual QString createLimitQueryForTable( const QString& schema, const QString& name, int limit = 10 );


That's also help remove the "this is gross" comment from the code...

Reply:
Thank you for the hint and the help! Anyway I think I'm not enough skilled to do such task. I will be pleased to close this PR in favour of a better one.

Reply:
@agiudiceandrea 

How's https://github.com/qgis/QGIS/pull/49950 ?

Reply:
Great! Thanks.